### ISSUE TITLE
Post Stats and Latest Post Summary do not show the correct numbers if the site timezone is in the next day

### ISSUE THREAD

**Issue by designsimply** [wordpress-mobile/WordPress-iOS#12499]
Steps to reproduce:

1. Go to My Site > $site > Settings > Time Zone.
1. Update the timezone to something several hours ahead of your device timezone.
1. Create and publish a new post on the site.
1. Visit the new post from a logged-out location to generate some views.
1. Wait for the views from the previous step to show up in Stats.
1. Wait until the timezone selected in the site settings is ahead of the device timezone date.<br><sup>For example, I am in UTC-6 and I selected UTC+10 so that the site settings date is the next day compared to my device date.</sup>
1. Go to Stats > Days and note that the views are showing for today.
1. Tap the title of the post to go to the individual Post Stats and note that no views are showing for today.
1. Go to Stats > Insights > Latest Post Summary and note that no views are showing for today.

**Note:** it's easier to test this using a test site with low traffic so you can easily see the changes in the numbers.

**Result:** Post Stats and Latest Post Summary do not show the correct numbers if the site timezone is in the next day compared to the current timezone.

<img src="https://user-images.githubusercontent.com/1119271/64555637-fb393c80-d2fa-11e9-981f-b550ad96beed.PNG" width="200" alt="IMG_3040"> <img src="https://user-images.githubusercontent.com/1119271/64555652-00968700-d2fb-11e9-807c-4db0cbff791d.PNG" width="200" alt="IMG_3044"> <img src="https://user-images.githubusercontent.com/1119271/64555650-00968700-d2fb-11e9-9474-8743ed64245e.PNG" width="200" alt="IMG_3043">
<sup>Tested with WPiOS 13.2 (App Store version) on iPhone 6S iOS 13.1 Beta 1.</sup>

**Comment by ScoutHarris** [comment]
@designsimply @danielebogo this is an issue with the endpoint - it doesn't honor the site timezone (Calypso has the same issue). I'll find the P2 about this and add it here for reference.

**Comment by staskus** [comment]
The issue is still reproducible, at least on the web:

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/b6339124-e1d5-4aee-bc68-a7d4daeaff20

While some pages show correct dates or indicate views, others show wrong dates or show no views for the post or for the current period. 

In all this time I don't remember seeing user reports about this particular issue. I'll timebox an hour to see if there's anything we could do from the app's side, or if the issue could only be solved by changes on the API side and close the issue.

**Comment by staskus** [comment]
The issue can be closed and is no longer reproducible on the app. The original issue was caused by the fact that `/sites/$site/stats/post` would return `date` in `UTC` rather than the site's timezone. Now this issue doesn't happen anymore. Example:

1. A post made April 3 GMT-10 (Honolulu) time `/sites/231579577/posts/10` timestamps the post correctly using `ISO8601` format: `2024-04-03T23:05:45-10:00`
2. `/sites/231579577/stats/post/10` correctly returns `date:"2024-04-03"`. Since the frontend clients are aware of this and operate in the site's timezone time, the correct requests return correct data.

However, there's an issue on the web with `Traffic` tab, which shows the chart for the current site's date but fetches data for the current device's date. I reported this issue:

- https://github.com/Automattic/wp-calypso/issues/89198
